src/main/scala/scala/compat/java8/converterImpl/MakesSteppers.scala

```diff
@@ -0,0 +1,160 @@
+trait MakesDoubleSeqStepper extends Any {
+  /** Generates a fresh stepper over `Double`s suitable for sequential use */
+  def stepper: DoubleStepper 
+}
+
+/** Classes or objects implementing this trait create `Double` steppers for map keys suitable for sequential use. */
+trait MakesDoubleKeySeqStepper extends Any {
+  /** Generates a fresh stepper over map keys of type `Double` suitable for sequential use */
+  def keyStepper: DoubleStepper 
+}
+
+/** Classes or objects implementing this trait create `Double` steppers  for map values suitable for sequential use. */
+trait MakesDoubleValueSeqStepper extends Any {
+  /** Generates a fresh stepper over map values of type `Double` suitable for sequential use */
+  def valueStepper: DoubleStepper 
+}
+
+/** Classes or objects implementing this trait create `Int` steppers suitable for sequential use. */
+trait MakesIntSeqStepper extends Any {
+  /** Generates a fresh stepper over `Int`s suitable for sequential use */
+  def stepper: IntStepper 
+}
+
+/** Classes or objects implementing this trait create `Int` steppers for map keys suitable for sequential use. */
+trait MakesIntKeySeqStepper extends Any {
+  /** Generates a fresh stepper over map keys of type `Int` suitable for sequential use */
+  def keyStepper: IntStepper 
+}
+
+/** Classes or objects implementing this trait create `Int` steppers for map values suitable for sequential use. */
+trait MakesIntValueSeqStepper extends Any {
+  /** Generates a fresh stepper over map values of type `Int` suitable for sequential use */
+  def valueStepper: IntStepper 
+}
+
+/** Classes or objects implementing this trait create `Long` steppers suitable for sequential use. */
+trait MakesLongSeqStepper extends Any {
+  /** Generates a fresh stepper over `Long`s suitable for sequential use */
+  def stepper: LongStepper 
+}
+
+/** Classes or objects implementing this trait create `Long` steppers for map keys suitable for sequential use. */
+trait MakesLongKeySeqStepper extends Any {
+  /** Generates a fresh stepper over map keys of type `Long` suitable for sequential use */
+  def keyStepper: LongStepper 
+}
+
+/** Classes or objects implementing this trait create `Long` steppers for map values suitable for sequential use. */
+trait MakesLongValueSeqStepper extends Any {
+  /** Generates a fresh stepper over map values of type `Long` suitable for sequential use */
+  def valueStepper: LongStepper 
+}
+
+/** Classes or objects implementing this trait create generic steppers suitable for sequential or parallel use. */
+trait MakesAnyStepper[A] extends Any {
+  /** Generates a fresh stepper over `A`s that can be efficiently subdivided */
+  def stepper: AnyStepper[A] with EfficientSubstep
+}
+
+/** Classes or objects implementing this trait create generic steppers for map keys suitable for sequential or parallel use. */
+trait MakesAnyKeyStepper[A] extends Any {
```

Comment:
So, the way this was supposed to work was that things that had `EfficientSubstep` could be run in a parallel stream just by asking for it, whereas if you don't have it you have to explicitly ask for a sequential stream and call the parallel method on it (or bite the bullet and turn it into an accumulator before you begin).  It was really for the stream layer, not for the steppers themselves.  And I had it as a marker trait so it can decorate both AnyStepper, which is inherently an iterator that implements tryAdvance and so on, and the spliterator-based steppers.  But I'm not sure I took advantage of that, so a redesign might be the way to go.

I am pretty sure that the Makes stuff was part of the type computation to hook up the right collections with the right streams, but otherwise the function idea is better.  I should check to make sure.